Assets/HoloToolkit/Input/Scripts/UI/Button.cs

```diff
@@ -0,0 +1,237 @@
+        {
+            get { return selected; }
+            set
+            {
+                if (selected != value)
+                {
+                    selected = value;
+                    UpdateButtonAnimation();
+                }
+            }
+        }
+
+        protected virtual void Awake()
+        {
+            if (focusedButtonId == 0)
+            {
+                focusedButtonId = Animator.StringToHash("Focused");
+            }
+
+            if (selectedButtonId == 0)
+            {
+                selectedButtonId = Animator.StringToHash("Selected");
+            }
+
+            if (deHydrateButtonId == 0)
+            {
+                deHydrateButtonId = Animator.StringToHash("Dehydrate");
+            }
+
+            if (stayFocusedButtonId == 0)
+            {
+                stayFocusedButtonId = Animator.StringToHash("StayFocused");
+            }
+        }
+
+        protected virtual void OnEnable()
+        {
+
+            // Set the initial alpha
+            if (ToolTipRenderer != null)
+            {
+                Color tipColor = ToolTipRenderer.material.color;
+                tipColor.a = 0.0f;
+                ToolTipRenderer.material.color = tipColor;
+                toolTipTimer = 0.0f;
+            }
+
+            UpdateVisuals();
+            UpdateButtonAnimation();
+        }
+
+        private void Update()
+        {
+            if (ToolTipRenderer != null && (Focused && toolTipTimer < ToolTipFadeTime) || (!Focused && toolTipTimer > 0.0f))
+            {
+                // Calculate the new time delta
+                toolTipTimer = toolTipTimer + (Focused ? Time.deltaTime : -Time.deltaTime);
+
+                // Stop the timer if it exceeds the limit.  Clamp doesn't work here since time can be outside the normal range in some situations
+                if (Focused && toolTipTimer > ToolTipFadeTime)
+                {
+                    toolTipTimer = ToolTipFadeTime;
+                }
+                else if (!Focused && toolTipTimer < 0.0f)
+                {
+                    toolTipTimer = 0.0f;
+                }
+
+                // Update the new opacity
+                if (ToolTipRenderer != null)
+                {
+                    Color tipColor = ToolTipRenderer.material.color;
+                    tipColor.a = Mathf.Clamp(toolTipTimer, 0, ToolTipFadeTime) / ToolTipFadeTime;
+                    ToolTipRenderer.material.color = tipColor;
+                }
+            }
+        }
+
+        public void DehydrateButton()
+        {
+            if (ButtonAnimator != null && ButtonAnimator.isInitialized)
+            {
+                var animatorHashes = ButtonAnimator.parameters;
+                for (int i = 0; i < animatorHashes.Length; i++)
+                {
+                    if (animatorHashes[i].nameHash == deHydrateButtonId)
+                    {
+                        ButtonAnimator.SetTrigger(deHydrateButtonId);
```

Comment:
It does not, but I don't think it should in this case, as this button is . The dehydration animation itself sets the button's renderer to inactive. I modified the animation so that it also sets the button collider to inactive, which I think addresses the issue.

That being said, I'm not convinced this button script is ready for main HoloToolkit yet. As such, I renamed it to TestButton and moved it to the Tests/Scripts location, so that it serves as an example script for the test scene, rather than something that people should start building on. Some design work would be required to make this into a great reusable button script that works across the board.